### PR TITLE
Miscellaneous Bug Fixes

### PR DESCRIPTION
**Pull request by jackthmp**
## Overview

Miscellaneous bug fixes related to user experience and banners.

## Changes Made

- Added animated bars for professors (for aestheticssss).
- Fixed constants and metrics for poll card UI.
- Sockets reconnect automatically 3 times before disconnecting upon errors.
- Fixed issues with persistent and unnecessary banners appearing. Moved logic to Socket.swift for consistency.
- Fixed crashing issue when simultaneously tapping on answer choices.
- Fixed issue where correct answer outline was reused when swiping between poll cards.

## Test Coverage

Manual testing

## Related PRs or Issues (delete if not applicable)

#365 
#376 
#381

### CODE REVIEW

File: Pollo/Views/Cards/PollOptionsCell.swift [hunk 1]
```diff
@@ -213,6 +213,7 @@ extension PollOptionsCell: MCChoiceSectionControllerDelegate {
             }
             // Select new choice
             let selectedIndex = adapter.section(for: sectionController)
+            guard selectedIndex < mcChoiceModels.count else { return }
```
Comment (jackthmp) [inline Pollo/Views/Cards/PollOptionsCell.swift]: #376

File: Pollo/Views/Cards/MCResultCell.swift [hunk 2]
```diff
@@ -200,6 +207,8 @@ class MCResultCell: UICollectionViewCell {
         
         highlightView.backgroundColor = .lightGrey
         highlightView.layer.borderColor = UIColor.coolGrey.cgColor
+        highlightView.layer.borderWidth = highlightViewBorderWidth
+        highlightView.layer.cornerRadius = highlightViewCornerRadius
```
Comment (jackthmp) [inline Pollo/Views/Cards/MCResultCell.swift]: #381

File: Pollo/Views/Cards/MCResultCell.swift [hunk 3]
```diff
@@ -187,8 +193,9 @@ class MCResultCell: UICollectionViewCell {
         dotView.isHidden = userRole == .admin
         
         containerView.backgroundColor = .clear
-        containerView.layer.borderWidth = highlightViewBorderWidth
-        containerView.layer.cornerRadius = highlightViewCornerRadius
+        containerView.layer.borderColor = UIColor.mediumGrey2.cgColor
```
Comment (jackthmp) [inline Pollo/Views/Cards/MCResultCell.swift]: #381

File: Pollo/Views/Cards/MCResultCell.swift [hunk 4]
```diff
@@ -108,9 +110,13 @@ class MCResultCell: UICollectionViewCell {
...
-            highlightView.snp.remakeConstraints { make in
-                make.leading.top.bottom.equalToSuperview()
-                make.width.equalToSuperview().multipliedBy(self.percentSelected)
+            UIView.animate(withDuration: 0.5) { [weak self] in
+                guard let self = self else { return }
+                self.highlightView.snp.remakeConstraints { make in
+                    make.leading.top.bottom.equalToSuperview()
+                    make.width.equalToSuperview().multipliedBy(self.percentSelected)
+                }
+                self.layoutIfNeeded()
```
Comment (jackthmp) [inline Pollo/Views/Cards/MCResultCell.swift]: Figured why not add some smooth animations lol

File: Pollo/Views/Cards/MCResultCell.swift [hunk 5]
```diff
@@ -31,13 +31,15 @@ class MCResultCell: UICollectionViewCell {
...
+    let containerViewBorderWidth: CGFloat = 0.6
+    let containerViewCorrectBorderWidth: CGFloat = 1.5
+    let containerViewCornerRadius: CGFloat = 8
     let containerViewHeight: CGFloat = 46
     let correctImageName = "correct"
     let dotViewBorderWidth: CGFloat = 2
     let dotViewLength: CGFloat = 23
-    let highlightViewBorderWidth: CGFloat = 0.6
+    let highlightViewBorderWidth: CGFloat = 0.3
+    let highlightViewCorrectBornerWidth: CGFloat = 1.0
```
Comment (chalo2000) [inline Pollo/Views/Cards/MCResultCell.swift]: highlightViewCorrectBor d erWidth

File: Pollo/Models/Socket.swift [hunk 6]
```diff
@@ -8,6 +8,7 @@
 
 import SocketIO
 import SwiftyJSON
+import NotificationBannerSwift
```
Comment (kevinchan159) [inline Pollo/Models/Socket.swift]: keep alphabetized

File: Pollo/Models/Socket.swift [hunk 7]
```diff
@@ -31,26 +32,35 @@ class Socket {
...
+            let banner = NotificationBanner.connectedBanner()
+            BannerController.shared.show(banner)
+
             self.delegate?.sessionConnected()
         }
         
         socket.on(clientEvent: .disconnect) { _, _ in
+            let banner = NotificationBanner.disconnectedBanner()
+            banner.onTap = { [weak self] in
+                guard let `self` = self else { return }
```
Comment (kevinchan159) [inline Pollo/Models/Socket.swift]: dont think you need the back ticks anymore?

File: Pollo/Models/Socket.swift [hunk 8]
```diff
@@ -104,6 +114,22 @@ class Socket {
         socket.connect()
     }
 
+    /// Manually reconnect socket to the server
+    func manualReconnect() {
+        let banner = NotificationBanner.reconnectingBanner()
+        BannerController.shared.show(banner)
+
+        socket.connect(timeoutAfter: 10) { [weak self] in
+            guard let `self` = self else { return }
```
Comment (kevinchan159) [inline Pollo/Models/Socket.swift]: same here

File: Pollo/Models/Socket.swift (hunk 7 above)
Comment (jackthmp) [reply Pollo/Models/Socket.swift]: you right, was just being consistent with the rest of the app lol

**Pull request merged**